images/fcos-bastion-image/root/usr/bin/fcos-download.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -x
+
+# download latest Fedora CoreOS
+mkdir -p ${TFTP_DIR}/fcos-{aarch64,x86_64}
+curl -o /tmp/stable.json https://builds.coreos.fedoraproject.org/streams/stable.json
+curl -o /tmp/fedora.gpg https://fedoraproject.org/fedora.gpg
+fcos_release=$(jq -r .architectures.aarch64.artifacts.metal.release /tmp/stable.json)
+echo -e "\n\e[31m~~ Fedora CoreOS Release ${fcos_release} ~~\e[0m\n"
+
+for arch in aarch64 x86_64; do
+  echo -e "\n----- $(date +"%Y-%m-%d %H:%M:%S") -----\n"
+  pushd ${TFTP_DIR}/fcos-${arch}
+  files_num=$(find ./ -name "*${fcos_release}*" -type f | wc -l)
+  if ((files_num < 9)); then
+    urls=$(jq .architectures.${arch}.artifacts.metal.formats.pxe /tmp/stable.json)
+    for pxeimg in kernel initramfs rootfs; do
+      curl -# -O $(jq -r .${pxeimg}.location <<< ${urls}) || echo "${pxeimg} download failed" || exit 1
```

Comment:
you sure the "|| .. || .." statement works as expected?


you could implement a function like

```bash

function fail() {
  echo "$1"
  exit 1
}
```


And use it as in `curl .... || fail "${pxeimg} download failed..."`

Reply:
Done